src/fidesops/ops/service/connectors/saas_query_config.py

```diff
@@ -106,28 +113,92 @@ def get_masking_request(self) -> Optional[SaaSRequest]:
     def generate_requests(
         self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
     ) -> List[SaaSRequestParams]:
-        """Takes the input_data and uses it to generate a list of SaaS request params"""
+        """
+        Takes the identity and reference values from input_data and combines them
+        with the connector_param values in use by the current request to generate
+        a list of request params.
+        """
 
-        filtered_data = self.node.typed_filtered_values(input_data)
+        current_request: Optional[SaaSRequest] = self.get_request_by_action("read")
+        if not current_request:
+            raise FidesopsException(
+                f"The 'read' action is not defined for the '{self.collection_name}' "
+                f"endpoint in {self.node.node.dataset.connection_key}"
+            )
 
         request_params = []
 
-        # Build SaaS requests for fields that are independent of each other
-        for string_path, reference_values in filtered_data.items():
-            for value in reference_values:
+        filtered_secrets = self._filtered_secrets(current_request)
+        grouped_inputs_list = input_data.pop(FIDESOPS_GROUPED_INPUTS, None)
+
+        # flatten the inputs
+        # list_ids: [[1,2,3]] -> list_ids: [1,2,3]
+        inputs_to_flatten = [
+            param_value.name
+            for param_value in current_request.param_values or []
+            if param_value.flatten
+        ]
+        for value in inputs_to_flatten:
+            input_data[value] = pydash.flatten(input_data.get(value))
+
+        # we want to preserve the grouped_input relationships so we take each
+        # individual group and generate the product with the ungrouped inputs
+        for grouped_inputs in grouped_inputs_list or [{}]:
+            param_value_maps = self._generate_product_list(
+                input_data, filtered_secrets, grouped_inputs
+            )
+            for param_value_map in param_value_maps:
                 request_params.append(
-                    self.generate_query({string_path: [value]}, policy)
+                    self.generate_query(
+                        {name: [value] for name, value in param_value_map.items()},
+                        policy,
+                    )
                 )
 
-        # Build SaaS requests for fields that are dependent on each other
-        grouped_input_data: List[Dict[str, Any]] = input_data.get(
-            FIDESOPS_GROUPED_INPUTS, []
-        )
-        for dependent_data in grouped_input_data:
-            request_params.append(self.generate_query(dependent_data, policy))
-
         return request_params
 
+    def _filtered_secrets(self, current_request: SaaSRequest) -> Dict[str, Any]:
```

Comment:
We don't want to create a product of reference values with connector params that are not used by the request, so we filter to only keep the connector params in use.